org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt

```diff
@@ -0,0 +1,131 @@
+package org.lflang.generator.ts
+
+import org.lflang.ErrorReporter
+import org.lflang.generator.FederateInstance
+import org.lflang.generator.PrependOperator
+import org.lflang.lf.Action
+import org.lflang.lf.Parameter
+import org.lflang.lf.Reactor
+import java.util.*
+
+class TSConstructorGenerator (
+    private val tsGenerator: TSGenerator,
+    private val errorReporter: ErrorReporter,
+    private val reactor : Reactor,
```

Comment:
Isn't the `Reactor` obtainable from the `FederateInstance`? It should be something like `getInstantiation().getReactorClass()`

Reply:
I think the reactor from federate.instantiation.reactor is the top-level reactor, and it could be different from the reactor being generated here (the reactor here could be any reactor inside the federate).

https://github.com/icyphy/lingua-franca/blob/master/org.lflang/src/org/lflang/generator/FederateInstance.xtend#L115

Reply:
Ah, I see, That makes sense.